sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -192,14 +207,51 @@ class SdkSynchronizer private constructor(
         }
     }
 
-    private val _status = MutableStateFlow<Synchronizer.Status>(DISCONNECTED)
+    private val _status = MutableStateFlow(DISCONNECTED)
 
-    var coroutineScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+    val coroutineScope: CoroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
 
     override val orchardBalances = processor.orchardBalances.asStateFlow()
     override val saplingBalances = processor.saplingBalances.asStateFlow()
     override val transparentBalance = processor.transparentBalance.asStateFlow()
 
+    private val refreshExchangeRateUsd = MutableSharedFlow<Unit>(replay = 1).apply { tryEmit(Unit) }
+
+    private var lastExchangeRateValue = ObserveFiatCurrencyResult()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    override val exchangeRateUsd =
+        channelFlow {
+            refreshExchangeRateUsd
+                .flatMapLatest {
+                    flow {
+                        emit(lastExchangeRateValue.copy(isLoading = true))
+                        lastExchangeRateValue =
+                            when (val result = fetchExchangeChangeUsd()) {
+                                is FetchFiatCurrencyResult.Error -> lastExchangeRateValue.copy(isLoading = false)
```

Comment:
yes it's supposed to be known and the staleness is handled by app

Reply:
We are allowing it to become excessively stale so that the app always has _something_. `lastExchangeRateValue` has a "time fetched" component, so the app always knows _how_ stale the value is. Nominally that's to enable the app to refetch as it desires, but it also lets the app decide at what point the rate is excessively stale _for its use case_.